pkg/osbuild/curl_source.go

```diff
@@ -3,14 +3,47 @@ package osbuild
 import (
 	"bytes"
 	"encoding/json"
+	"regexp"
+
+	"github.com/osbuild/images/pkg/rpmmd"
 )
 
+var curlDigestPattern = regexp.MustCompile(`(md5|sha1|sha256|sha384|sha512):[0-9a-f]{32,128}`)
+
 type CurlSource struct {
 	Items map[string]CurlSourceItem `json:"items"`
 }
 
 func (CurlSource) isSource() {}
 
+func NewCurlSource() *CurlSource {
+	return &CurlSource{
+		Items: make(map[string]CurlSourceItem),
+	}
+}
+
+func NewCurlPackageItem(pkg rpmmd.PackageSpec) CurlSourceItem {
+	if !curlDigestPattern.MatchString(pkg.Checksum) {
+		panic("curl package source item has invalid digest")
+	}
+	item := new(CurlSourceOptions)
+	item.URL = pkg.RemoteLocation
+	if pkg.Secrets == "org.osbuild.rhsm" {
+		item.Secrets = &URLSecrets{
+			Name: "org.osbuild.rhsm",
+		}
+	}
+	item.Insecure = pkg.IgnoreSSL
+	return item
+}
+
+// AddPackage adds a pkg to the curl source to download. Will panic if any of
+// the supplied options are invalid or missing.
+func (source *CurlSource) AddPackage(pkg rpmmd.PackageSpec) {
+	item := NewCurlPackageItem(pkg)
+	source.Items[pkg.Checksum] = item
+}
+
```

Comment:
minor nitpick: while reading the file, I was searching for `CurlSourceItem` definition while reading the `CurlSource` definition. I would suggest to move the whole block below `CurlSourceItem` definition, so that the code is easy to read from the top down...

Reply:
Done.